### PR TITLE
merge common-public into common, remove react-xnft, react-xnft-dom-renderer

### PR DESCRIPTION
**Pull request by tomlinton**
- Move all the `common-public` code into `common`
- Remove `react-xnft`
- Remove `react-xnft-dom-renderer`
- Remove most the example xNFTS, except for a few important ones `explorer`, `prices`, `provider-test`, `simulator`
- Use the published versions of `react-xnft` for the remaining xNFTs
- Remove the legacy option from `xnft-cli`
- Regenerate the root `yarn.lock`

### CODE REVIEW

**Comment by wentokay** [comment]
`examples/xnft/iframe` is currently used by xnft-cli https://github.com/coral-xyz/backpack/blob/f7fcb65b267f2abbd18160acec9d059d76fe9f5c/packages/xnft-cli/package.json#L8-L9

**Comment by wentokay** [comment]
kinda related https://github.com/coral-xyz/backpack/issues/2877#issuecomment-1438997257

**Comment by tomlinton** [comment]
> `examples/xnft/iframe` is currently used by xnft-cli https://github.com/coral-xyz/backpack/blob/f7fcb65b267f2abbd18160acec9d059d76fe9f5c/packages/xnft-cli/package.json#L8-L9

This PR pulls out those package.json scripts and removes the legacy stuff from xnft-cli, is it still needed?

**Comment by wentokay** [comment]
> > `examples/xnft/iframe` is currently used by xnft-cli https://github.com/coral-xyz/backpack/blob/f7fcb65b267f2abbd18160acec9d059d76fe9f5c/packages/xnft-cli/package.json#L8-L9
> 
> This PR pulls out those package.json scripts and removes the legacy stuff from xnft-cli, is it still needed?

it's used in https://github.com/coral-xyz/xnft-quickstart/blob/7b1144d044890650b4416c661fc18a799549102b/package.json#L6

@ph101pp is there a new alternative approach to `--iframe` that's not being used here?

File: packages/xnft-cli/index.js [hunk 1]
```diff
@@ -5,16 +5,12 @@
 const { program } = require("commander");
 const native = require("./native");
 const fs = require("fs");
-const legacy = require("./legacy");
 const bundle = require("./new");
 
 const pkg = JSON.parse(fs.readFileSync(__dirname + "/package.json").toString());
 program.version(pkg.version);
 
 native(program.command("native"));
```
Comment (ph101pp) [inline packages/xnft-cli/index.js:13]: the `native` command could be removed too

**Comment by ph101pp** [comment]
iframe example and all dependencies from xnft-cli can safely be removed

> it's used in https://github.com/coral-xyz/xnft-quickstart/blob/7b1144d044890650b4416c661fc18a799549102b/package.json#L6

This is using the a new version of the `--iframe` option from here: https://github.com/coral-xyz/backpack/blob/master/packages/xnft-cli/new/dev/index.js.

File: packages/xnft-cli/index.js (hunk 1 above)
Comment (tomlinton) [reply packages/xnft-cli/index.js]: Great, pulled it out

**Pull request merged**